src/ring_theory/nilpotent.lean

```diff
@@ -44,6 +44,10 @@ end
 @[simp] lemma is_nilpotent_neg_iff [ring R] : is_nilpotent (-x) ↔ is_nilpotent x :=
 ⟨λ h, neg_neg x ▸ h.neg, λ h, h.neg⟩
 
+lemma is_nilpotent.map [monoid_with_zero R] [monoid_with_zero S] {r : R}
+  (hr : is_nilpotent r) (f : monoid_with_zero_hom R S) : is_nilpotent (f r) :=
```

Comment:
I think this should take advantage of the new `fun_like` classes, so that it works for `monoid_with_zero_hom` and `ring_hom` simultaneously.

@Vierkantor you are the expert. Do you agree?